### PR TITLE
Refactoring DSL

### PR DESCRIPTION
**Pull request by jexp**
* Split convert method into steps
* Config class
* Context class for table information

### CODE REVIEW

File: src/main/java/org/neo4j/sql2cypher/SQLToCypher.java [hunk 1]
```diff
@@ -44,38 +44,87 @@
...
+        private final Map<Table<?>, Node> tables = new HashMap<>();
+
+        void addTable(Table<?> table, Node node) {
+            this.tables.put(table,node);
+        }
+
+        private Node getNode(Table<?> t) {
+            return tables.get(t);
+        }
+    }
```
Comment (lukaseder) [inline src/main/java/org/neo4j/sql2cypher/SQLToCypher.java:75]: Is extracting this `Context` truly better? The jOOQ parser had this before. And then *all* the 581734 parser methods had to pass around `Context`, when in fact, the context could have been just `this`.

File: src/main/java/org/neo4j/sql2cypher/SQLToCypher.java [hunk 2]
```diff
@@ -44,38 +44,87 @@
...
-        return renderer.render(statement(query));
+        Context ctx = new Context();
+        Select<?> query = parse(sql);
+        ResultStatement statement = statement(query, ctx);
+        return render(statement);
     }
 
-    private ResultStatement statement(Select<?> x) {
+    Select<?> parse(String sql) {
+        Parser parser = DSL.using(config.getDialect()).parser();
```
Comment (lukaseder) [inline src/main/java/org/neo4j/sql2cypher/SQLToCypher.java:113]: In the context of this utility, the jOOQ `SQLDialect` won't be too useful (except for debugging, perhaps?) as it only affects how jOOQ *generates* SQL. What you want to set here is the `Settings.parseDialect`:

```java
DSL.using(dialect, new Settings().withParseDialect(dialect))
```

File: src/main/java/org/neo4j/sql2cypher/SQLToCypher.java [hunk 3]
```diff
@@ -44,38 +44,87 @@
...
+        return renderer.render(statement);
+    }
+
+    ResultStatement statement(Select<?> x, Context ctx) {
 
         // Done lazy as otherwise the property containers won't be resolved
-        Supplier<List<Expression>> resultColumnsSupplier = () -> x.$select().stream().map(t -> (Expression) expression(t)).toList();
+        Supplier<List<Expression>> resultColumnsSupplier =
+                () -> x.$select().stream()
+                        .map(t -> (Expression) expression(t, ctx)).toList();
```
Comment (lukaseder) [inline src/main/java/org/neo4j/sql2cypher/SQLToCypher.java:127]: Just a question, has this auto formatting been checked in, or how do you folks usually work? I don't have an opinion, other than wanting to adapt to your coding style.

Comment (jexp) [reply src/main/java/org/neo4j/sql2cypher/SQLToCypher.java:127]: Multi line is kinda easier to read.

File: src/main/java/org/neo4j/sql2cypher/SQLToCypher.java (hunk 2 above)
Comment (jexp) [reply src/main/java/org/neo4j/sql2cypher/SQLToCypher.java:113]: Ok, added it in the new version.

**Pull request closed without merging**